Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/StorageReset.kt

```diff
@@ -0,0 +1,18 @@
+package de.rki.coronawarnapp.storage
+
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import de.rki.coronawarnapp.util.datastore.clear
+import de.rki.coronawarnapp.util.reset.Resettable
+import timber.log.Timber
+import javax.inject.Inject
+
+class StorageReset @Inject constructor(
+    @StorageDataStore private val dataStore: DataStore<Preferences>
+) : Resettable {
```

Comment:
what is the purpose of this class? can't we just clear `dataStore` directly ?

Reply:
ok, but you would need to create a class for every DataStore then , right ? 
if takes specific datastore because of the the annotation

Reply:
So, your idea is to create a single class where every datastore gets injected and resetted?

Reply:
No, you can directly make the respective DataStore resttable and just do the logic there, instead of creating a specific class for each datastore to reset other the datastore, maybe you can highlight why the approach you follow here is better ?

Reply:
how can I do that?

Reply:
I see why you introduced this class, it is because you want to use the same Datastore for three settings ,so instead of resetting each and evert settings which at the end resets the same DataStore.  It is is confusing and I think you should be keep them saved in different files which is exactly my point above https://github.com/corona-warn-app/cwa-app-android/pull/5236#discussion_r884651509

Reply:
In my opinion it's not really confusing and actually simpler than creating a distinct datastore for every setting which is part of the storage module and then make every setting resettable to clear their own datastore.

Reply:
See https://github.com/corona-warn-app/cwa-app-android/pull/5236#discussion_r893471922

Reply:
See https://github.com/corona-warn-app/cwa-app-android/pull/5236#discussion_r894230617